lib/root/root_page.dart

```diff
@@ -19,31 +23,36 @@ class RootPage extends StatelessWidget {
       builder: (context, child) {
         final tabsRouter = AutoTabsRouter.of(context);
 
-        return Scaffold(
-          body: child,
-          bottomNavigationBar: NavigationBar(
-            selectedIndex: tabsRouter.activeIndex,
-            onDestinationSelected: (index) {
-              tabsRouter.setActiveIndex(index);
-            },
-            destinations: const [
-              NavigationDestination(
-                icon: Icon(LineAwesome.home_solid),
-                label: 'Home',
-              ),
-              NavigationDestination(
-                icon: Icon(LineAwesome.tag_solid),
-                label: 'Shop',
-              ),
-              NavigationDestination(
-                icon: Icon(LineAwesome.user_solid),
-                label: 'Account',
-              ),
-              NavigationDestination(
-                icon: Icon(LineAwesome.shopping_bag_solid),
-                label: 'Cart',
-              ),
-            ],
+        return BlocProvider<CartIconCubit>(
+          create: (context) => CartIconCubit(
+            globalCartBloc: context.read<GlobalCartBloc>(),
+          ),
+          child: Scaffold(
+            body: child,
+            bottomNavigationBar: NavigationBar(
+              selectedIndex: tabsRouter.activeIndex,
+              onDestinationSelected: (index) {
+                tabsRouter.setActiveIndex(index);
+              },
+              destinations: const [
+                NavigationDestination(
+                  icon: Icon(LineAwesome.home_solid),
+                  label: 'Home',
+                ),
+                NavigationDestination(
+                  icon: Icon(LineAwesome.tag_solid),
+                  label: 'Shop',
+                ),
+                NavigationDestination(
+                  icon: Icon(LineAwesome.user_solid),
+                  label: 'Account',
+                ),
+                NavigationDestination(
+                  icon: CartIcon(),
```

Comment:
How about putting the `BlocProvider<CartIconCubit>` around this small widget instead of the whole page?